includes/classes/Indexable.php

```diff
@@ -819,11 +826,44 @@ abstract public function build_mapping();
 	abstract public function build_settings();
 
 	/**
-	 * Must implement a method that handles sending mapping to ES
+	 * Retrieve index settings from ES
+	 *
+	 * @since 3.6
+	 * @return array
+	 */
+	public function get_index_settings() {
+		$index_name = $this->get_index_name();
+
+		$result = Elasticsearch::factory()->get_index_settings( $index_name );
+
+		if ( is_wp_error( $result ) ) {
+			return $result;
+		}
+
+		// The ES response has the data nested under $index_name, so pull that out so we match the format of
+		// the settings from Indexable::build_settings() and the mapping file, for consistency
+		if ( ! isset( $result[ $index_name ]['settings'] ) ) {
+			return new \WP_Error( 'index-settings-not-found', sprintf( 'Index settings for %s were not found in the Elasticsearch response', $index_name ) );
+		}
```

Comment:
Would it make sense to extract this directly in `get_index_settings()` in `Elasticsearch.php`? This way both the response to `get_index_settings()` and the argument to `update_index_settings()` in that class would be consistent.

Unless it's a common practice in `Elasticsearch.php` to return a response as similar as possible to the one obtained from ES, in which case it would make sense to do it here.

Reply:
@netsuso there is a mixture of both approaches in `Elasticsearch` - for this, it probably makes sense to leave it returning the raw response, because the index name doesn't necessarily point to a single index - you could get a few with `index-1,index2`, or use a wildcard `index-*`, or an index alias, all which could return more than 1 index.

Reply:
Oh, I hadn't noticed the possibility of multiple indexes, then it makes sense doing it this way :)